media/client/ipc/source/MediaPipelineIpc.cpp

```diff
@@ -226,53 +226,90 @@ bool MediaPipelineIpc::attachSource(const std::unique_ptr<IMediaPipeline::MediaS
     if (configType == SourceConfigType::VIDEO_DOLBY_VISION || configType == SourceConfigType::VIDEO ||
         configType == SourceConfigType::AUDIO)
     {
-        IMediaPipeline::MediaSourceAV &mediaSourceAV = dynamic_cast<IMediaPipeline::MediaSourceAV &>(*source);
-        request.set_segment_alignment(convertSegmentAlignment(mediaSourceAV.getSegmentAlignment()));
+        IMediaPipeline::MediaSourceAV *mediaSourceAV = nullptr;
+        try
+        {
+            mediaSourceAV = &dynamic_cast<IMediaPipeline::MediaSourceAV &>(*source);
+        }
+        catch (const std::bad_cast &e)
+        {
+            RIALTO_CLIENT_LOG_ERROR("Failed to get the audio video source, reason: %s", e.what());
+        }
+
+        request.set_segment_alignment(convertSegmentAlignment(mediaSourceAV->getSegmentAlignment()));
 
-        if (mediaSourceAV.getCodecData())
+        if (mediaSourceAV->getCodecData())
         {
-            request.mutable_codec_data()->set_data(mediaSourceAV.getCodecData()->data.data(),
-                                                   mediaSourceAV.getCodecData()->data.size());
-            request.mutable_codec_data()->set_type(convertCodecDataType(mediaSourceAV.getCodecData()->type));
+            request.mutable_codec_data()->set_data(mediaSourceAV->getCodecData()->data.data(),
+                                                   mediaSourceAV->getCodecData()->data.size());
+            request.mutable_codec_data()->set_type(convertCodecDataType(mediaSourceAV->getCodecData()->type));
         }
-        request.set_stream_format(convertStreamFormat(mediaSourceAV.getStreamFormat()));
+        request.set_stream_format(convertStreamFormat(mediaSourceAV->getStreamFormat()));
 
         if (configType == SourceConfigType::VIDEO_DOLBY_VISION)
         {
-            IMediaPipeline::MediaSourceVideoDolbyVision &mediaSourceDolby =
-                dynamic_cast<IMediaPipeline::MediaSourceVideoDolbyVision &>(*source);
+            try
+            {
+                IMediaPipeline::MediaSourceVideoDolbyVision &mediaSourceDolby =
+                    dynamic_cast<IMediaPipeline::MediaSourceVideoDolbyVision &>(*source);
 
-            request.set_width(mediaSourceDolby.getWidth());
-            request.set_height(mediaSourceDolby.getHeight());
-            request.set_dolby_vision_profile(mediaSourceDolby.getDolbyVisionProfile());
+                request.set_width(mediaSourceDolby.getWidth());
+                request.set_height(mediaSourceDolby.getHeight());
+                request.set_dolby_vision_profile(mediaSourceDolby.getDolbyVisionProfile());
+            }
+            catch (const std::bad_cast &e)
+            {
+                RIALTO_CLIENT_LOG_ERROR("Failed to get the video dolby vision media source, reason: %s", e.what());
```

Comment:
So you want still continue with attaching the source here? It will mean that, we will attach "half-created" source and this might lead to bugs, which will be annoying to debug.
I would fail attaching the source when cast fails.